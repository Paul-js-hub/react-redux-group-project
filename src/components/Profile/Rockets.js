import Card from 'react-bootstrap/Card';

const Rockets = () => (
  <>
    <Card body>This is some text within a card body.</Card>
  </>
);

export default Rockets;
