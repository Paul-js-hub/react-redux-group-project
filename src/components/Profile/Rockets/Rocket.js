import PropTypes from 'prop-types';
import Card from 'react-bootstrap/Card';

const Rockets = (props) => {
  const { name } = props;

  return (
    <Card body>
      {name}
    </Card>
  );
};

Rockets.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Rockets;
