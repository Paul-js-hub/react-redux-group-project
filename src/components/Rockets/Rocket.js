import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import style from './rockets.module.css';

const Rocket = (props) => {
  const {
    id, image, name, desc, handleClick, reserved,
  } = props;

  return (
    <Row className="align-items-center my-5">
      <Col xl={3} md={5}>
        <div className="rocketImage">
          <img src={image} alt={name} width="100%" />
        </div>
      </Col>
      <Col>
        <div className={style.content}>
          <h2>{name}</h2>
          <p className={style.description}>
            {reserved && (<Badge className={style.badge} bg="success">Reserved</Badge>)}
            {desc}
          </p>
        </div>
        {/* {reserveBtn(reserved)} */}
        <Button
          type="button"
          onClick={() => handleClick(id)}
          variant={reserved ? ('secondary') : ('primary')}
          className={style.btn}
        >
          {reserved ? ('Cancel Reservation') : ('Reserve Rocket')}
        </Button>
      </Col>
      {/* {reserved ? ('hello')} */}
    </Row>
  );
};

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Rocket;
