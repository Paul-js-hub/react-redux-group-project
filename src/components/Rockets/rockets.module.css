.content {
  position: relative;
}

.description .badge {
  /* animation-name: showBadge;
  animation-duration: 0.3s; */
  transition: all 0.5s ease-in-out;
  width: 0;
  opacity: 0;
  overflow: hidden;
  padding: 0;
}

.description .badge.badgeShow {
  width: 69.23px;
  opacity: 1;
  margin-right: 10px;
  visibility: visible;
  padding: .35em .65em;
}


.btn {
  min-width: 165px;
}

.fadeIn {
  transition: all 0.1s ease-in-out;
}

.description {
  min-height: 8rem;
}

@keyframes showBadge {
  from {
    width: 0;
  }

  to {
    width: 69.23px;
  }
}
