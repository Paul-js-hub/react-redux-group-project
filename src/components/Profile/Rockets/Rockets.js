import PropTypes from 'prop-types';
import Rocket from './Rocket';

const Rockets = (props) => {
  const { reserved } = props;
  return (
    <>
      {reserved.map((rocket) => (
        <Rocket
          key={rocket.id}
          name={rocket.name}
        />
      ))}
    </>
  );
};

Rockets.propTypes = {
  reserved: PropTypes.instanceOf(Array).isRequired,
};

export default Rockets;
