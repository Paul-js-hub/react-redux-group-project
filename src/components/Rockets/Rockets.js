import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Container from 'react-bootstrap/Container';
import fetchRockets from '../../redux/rockets/RocketsService';
import { reserveRocket } from '../../redux/rockets/rockets';
import Rocket from './Rocket';

const Rockets = () => {
  const dispatch = useDispatch();
  const { loaded, data } = useSelector((state) => state.rockets);
  useEffect(() => {
    if (!loaded) {
      dispatch(fetchRockets());
    }
  }, []);

  const reserve = (id) => {
    dispatch(reserveRocket(id));
  };
  return (
    <Container fluid>
      {data.map((rocket) => (
        <Rocket
          key={rocket.id}
          id={rocket.id}
          image={rocket.flickr_images[0]}
          name={rocket.name}
          desc={rocket.description}
          reserved={rocket.reserved}
          handleClick={reserve}
        />
      ))}
    </Container>
  );
};

export default Rockets;
