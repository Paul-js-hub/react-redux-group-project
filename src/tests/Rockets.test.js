import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import Rockets from '../Rockets';

describe('Rockets test', () => {
  it('Rockets render correctly', () => {
    const RocketsPage = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(RocketsPage).toMatchSnapshot();
  });

  test('Render Rockets component', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
  });
});
