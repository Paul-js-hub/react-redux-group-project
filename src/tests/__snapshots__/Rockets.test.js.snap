// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets test Rockets render correctly 1`] = `
<div
  className="container"
/>
`;
